app/src/main/kotlin/com/intive/picover/validators/TextValidator.kt

```diff
@@ -7,33 +7,26 @@ import com.intive.picover.validators.ValidationStatus.ValidText
 
 class TextValidator private constructor(
 	private val text: String,
-	private val allowEmpty: Boolean?,
-	private val allowBlank: Boolean?,
+	private val allowEmpty: Boolean,
+	private val allowBlank: Boolean,
 	private val maxLength: Int?,
 ) {
 
-	fun validatingText(): ValidationStatus {
-		if (allowEmpty == false && text.isEmpty()) {
-			return EmptyText
-		}
-		if (allowBlank == false && text.isBlank()) {
-			return BlankText
-		}
-		if (maxLength != null && text.length > maxLength) {
-			return TooLongText
-		}
-		return ValidText
+	fun validatingText() = when {
+		!allowEmpty && text.isEmpty() -> EmptyText
+		!allowBlank && text.isBlank() -> BlankText
+		maxLength != null && text.length > maxLength -> TooLongText
+		else -> ValidText
 	}
 
 	class Builder(private val text: String) {
-		private var allowEmpty: Boolean? = null
-		private var allowBlank: Boolean? = null
+		private var allowEmpty = true
+		private var allowBlank = true
 		private var maxLength: Int? = null
 
 		fun allowEmpty(allow: Boolean) = apply { allowEmpty = allow }
 		fun allowBlank(allow: Boolean) = apply { allowBlank = allow }
 		fun maxLength(length: Int) = apply { maxLength = length }
```

Comment:
Could you give me a more detailed example because I don't understand? Do you mean to change the Builder to allow creating it like this?: 
```
Textalidator.Builder.build {
    allowEmpty = true
    allowBlank = false
    maxLength = 4
}
```